components/ws-proxy/pkg/proxy/workspacerouter.go

```diff
@@ -51,15 +52,22 @@ func HostBasedRouter(header, wsHostSuffix string) WorkspaceRouter {
 			}
 			blobserveRouter = r.MatcherFunc(matchBlobserveHostHeader(wsHostSuffix, getHostHeader)).Subrouter()
 			portRouter      = r.MatcherFunc(matchWorkspacePortHostHeader(wsHostSuffix, getHostHeader)).Subrouter()
-			theiaRouter     = r.MatcherFunc(matchWorkspaceHostHeader(wsHostSuffix, getHostHeader)).Subrouter()
+
+			// Allow to match all cluster prefixes (ws-eu01, ws-eu01, ws-us01, ...)
+			// independent of the prefix of the cluster this component is
+			// deployed to. If the URL has an older WS cluster prefix that does
+			// not exist anymore, the workspaceMustExistHandler function in
+			// routes.go redirects to /start/.
+			allClusterWsHostSuffixRegex = fmt.Sprintf(`(.ws[a-zA-Z0-9-]*)?.%s`, wsHostName)
```

Comment:
This ties the domain structure down which is a stronger requirement than what we've had so far.
Until now the only requirement was that the workspace domains have to be subdomains of the Gitpod installation as a whole. Now they also need to start with `ws`.

Is there a way we could do away with this restriction, i.e. not start imposing it in the first place?

Reply:
Yes, that's right. I came from the direction that our helm chart always generates a workspace domain starting with `ws`.

https://github.com/gitpod-io/gitpod/blob/7d0986778e5294bf571c4e96ac013bf46f1319ca/chart/templates/ws-proxy-configmap.yaml#L52

---

What we can do is to move the regex as an optional value to the config file resp. `values.yaml` file and use as default the value of [`workspaceHostSuffix`](https://github.com/gitpod-io/gitpod/blob/7d0986778e5294bf571c4e96ac013bf46f1319ca/chart/templates/ws-proxy-configmap.yaml#L52) (as we do now). That would allow us to configure this for our Gitpod SaaS deployment only (in gitpod-com) and change the regex at any time so that it matches all our workspace clusters.

What do you think, @csweichel?

Reply:
> Is there a way we could do away with this restriction, i.e. not start imposing it in the first place?

In general we always had a restriction that we generate in the chart: Up until now it was a suffix match, now a regex which Cornelius proposes to generate and inject from the chart as well.

But it indeed feels like a tad to complex. What if we had no restriction on the 2nd level at all, e.g. use `[^\.]+` instead of `ws[a-zA-Z0-9-]*`? IMO even the check before was too much, and served no purpose beyond a gut feeling of "these request should not go here".

Reply:
I reckon the less restriction/need for configuration we build in here the better.
If we can get away with something like `[^\.]+` we should do that.